src/decode.rs

```diff
@@ -4765,7 +4763,7 @@ unsafe fn decode_b(
     } else {
         b.skip_mode = 0 as libc::c_int as uint8_t;
     }
-    if b.skip_mode as libc::c_int != 0 || !seg.is_null() && (*seg).skip != 0 {
+    if b.skip_mode as libc::c_int != 0 || seg.map(|seg| seg.skip != 0).unwrap_or_default() {
```

Comment:
If we have `.unwrap_or(true)`s, then I think we should use `.unwrap_or(false)` here.

Reply:
Same for the others.

Reply:
Done!